src/operator/tensor/indexing_op.h

```diff
@@ -312,15 +407,7 @@ void TakeOpBackward(const nnvm::NodeAttrs& attrs,
             if ((grad_out.shape_[0] < grad_out.shape_[1]) && (grad_out.shape_[0] < 512)) {
```

Comment:
unsigned long long shape_in_prod = 
        static_cast<unsigned long long>(grad_in.shape_[0])*
        static_cast<unsigned long long>(grad_in.shape_[1]);

      unsigned long long shape_out_prod = 
        static_cast<unsigned long long>(grad_out.shape_[0])*
        static_cast<unsigned long long>(grad_out.shape_[1]);

      if (shape_out_prod < 16384ULL && shape_in_prod < 16384ULL) {
        AddTakeGrad(grad_in, data, grad_out);

This is a new criteria I came up with. Should I put it in a separate PR?